baseplate.go

```diff
@@ -301,15 +314,22 @@ type NewArgs struct {
 // run your service on.  The returned context will be cancelled when the
 // Baseplate is closed.
 func New(ctx context.Context, args NewArgs) (context.Context, Baseplate, error) {
-	var cfger Configer
-	if args.ServiceCfg != nil {
-		cfger = args.ServiceCfg
-	} else {
-		cfger = new(Config)
+	if args.ConfigPath == "" && args.ServiceCfg == nil {
+		return nil, nil, errors.New("baseplate.New: at least one of ConfigPath and ServiceCfg must be set")
 	}
-	err := parseConfig(args.ConfigPath, cfger)
-	if err != nil {
-		return nil, nil, err
+
+	cfger := args.ServiceCfg
+	if args.ConfigPath != "" {
+		if cfger == nil {
```

Comment:
I think it's clearer to put this outside the configPath != "" check.  It requires a lot of squinting and thinking to prove to oneself that line 334 cannot panic because cfger can't be nil at that point.

Checking after line 321 makes it very clear and ensures that a future change to line 317 or nearby can't introduce that as a bug.

Reply:
on second thought I should remove `args.ConfigPath` instead of deprecating it, so this will be irrelevant.